src/workflow_provider/default_vm_workflow_provider.cpp

```diff
@@ -0,0 +1,263 @@
+            }
+        }
+    }
+
+    return workflows_map;
+}
+} // namespace
+
+mp::DefaultVMWorkflowProvider::DefaultVMWorkflowProvider(const QUrl& workflows_url, URLDownloader* downloader,
+                                                         const QDir& archive_dir,
+                                                         const std::chrono::milliseconds& workflows_ttl)
+    : workflows_url{workflows_url},
+      url_downloader{downloader},
+      archive_file_path{archive_dir.filePath(github_workflows_archive_name)},
+      workflows_ttl{workflows_ttl}
+{
+    update_workflows();
+}
+
+mp::DefaultVMWorkflowProvider::DefaultVMWorkflowProvider(URLDownloader* downloader, const QDir& archive_dir,
+                                                         const std::chrono::milliseconds& workflows_ttl)
+    : DefaultVMWorkflowProvider(default_workflow_url, downloader, archive_dir, workflows_ttl)
+{
+}
+
+mp::Query mp::DefaultVMWorkflowProvider::fetch_workflow_for(const std::string& workflow_name,
+                                                            VirtualMachineDescription& vm_desc)
+{
+    update_workflows();
+
+    Query query{"", "", false, "", Query::Type::Alias};
+    auto& config = workflow_map.at(workflow_name);
+    auto workflow_config = YAML::Load(config);
+
+    auto workflow_instance = workflow_config["instances"][workflow_name];
+
+    if (workflow_instance["image"])
+    {
+        // TODO: Support http later.
+        // This only supports the "alias" and "remote:alias" scheme at this time
+        auto image_str{workflow_config["instances"][workflow_name]["image"].as<std::string>()};
+        auto tokens = mp::utils::split(image_str, ":");
+
+        if (tokens.size() == 2)
+        {
+            query.remote_name = tokens[0];
+            query.release = tokens[1];
+        }
+        else if (tokens.size() == 1)
+        {
+            query.release = tokens[0];
+        }
+        else
+        {
+            throw InvalidWorkflowException("Unsupported image scheme in Workflow");
+        }
+    }
+
+    if (workflow_instance["limits"]["min-cpu"])
+    {
+        try
+        {
+            auto min_cpus = workflow_instance["limits"]["min-cpu"].as<int>();
+
+            if (vm_desc.num_cores == 0)
+            {
+                vm_desc.num_cores = min_cpus;
+            }
+            else if (vm_desc.num_cores < min_cpus)
+            {
+                throw WorkflowMinimumException("Number of CPUs", std::to_string(min_cpus));
+            }
+        }
+        catch (const YAML::BadConversion&)
+        {
+            throw InvalidWorkflowException(fmt::format("Minimum CPU value in workflow is invalid"));
+        }
+    }
+
+    if (workflow_instance["limits"]["min-mem"])
+    {
+        auto min_mem_size_str{workflow_instance["limits"]["min-mem"].as<std::string>()};
+
+        try
+        {
+            MemorySize min_mem_size{min_mem_size_str};
+
+            if (vm_desc.mem_size.in_bytes() == 0)
```

Comment:
So yeah, I'd be happier with `.empty()` or so here.

Reply:
Ack

Reply:
So given that I'm reverting the `MemorySize` change back to the original, then we'll have to leave this be in order to determine if a user provided a memory size option on the `launch` cli.  If a user really did specify 0 bytes on the cli, that will be caught like it is currently.